flink-ml-lib/src/main/java/org/apache/flink/ml/common/gbt/GBTRunner.java

```diff
@@ -0,0 +1,304 @@
+                                    if (node.isLeaf) {
+                                        continue;
+                                    }
+                                    featureImportanceMap.merge(
+                                            node.split.featureId, node.split.gain, Double::sum);
+                                    sum += node.split.gain;
+                                }
+                            }
+                            if (sum > 0.) {
+                                for (Map.Entry<Integer, Double> entry :
+                                        featureImportanceMap.entrySet()) {
+                                    entry.setValue(entry.getValue() / sum);
+                                }
+                            }
+
+                            List<String> featureNames = value.featureNames;
+                            return featureImportanceMap.entrySet().stream()
+                                    .collect(
+                                            Collectors.toMap(
+                                                    d -> featureNames.get(d.getKey()),
+                                                    Map.Entry::getValue));
+                        })
+                .returns(Types.MAP(Types.STRING, Types.DOUBLE));
+    }
+
+    private static DataStream<GBTModelData> boost(
+            Table dataTable,
+            BoostingStrategy strategy,
+            DataStream<FeatureMeta> featureMeta,
+            DataStream<Tuple2<Double, Long>> labelSumCount) {
+        StreamTableEnvironment tEnv =
+                (StreamTableEnvironment) ((TableImpl) dataTable).getTableEnvironment();
+
+        final String featureMetaBcName = "featureMeta";
+        final String labelSumCountBcName = "labelSumCount";
+        Map<String, DataStream<?>> bcMap = new HashMap<>();
+        bcMap.put(featureMetaBcName, featureMeta);
+        bcMap.put(labelSumCountBcName, labelSumCount);
+
+        DataStream<TrainContext> initTrainContext =
+                BroadcastUtils.withBroadcastStream(
+                        Collections.singletonList(
+                                tEnv.toDataStream(tEnv.fromValues(0), Integer.class)),
+                        bcMap,
+                        (inputs) -> {
+                            //noinspection unchecked
+                            DataStream<Integer> input = (DataStream<Integer>) (inputs.get(0));
+                            return input.map(
+                                    new InitTrainContextFunction(
+                                            featureMetaBcName, labelSumCountBcName, strategy));
+                        });
+
+        DataStream<Row> data = tEnv.toDataStream(dataTable);
+        DataStreamList dataStreamList =
+                Iterations.iterateBoundedStreamsUntilTermination(
+                        DataStreamList.of(initTrainContext.broadcast()),
+                        ReplayableDataStreamList.notReplay(data, featureMeta),
```

Comment:
The `featureMeta` seems never used later. Can you remove it?

Reply:
No problem.